client/core/core.go

```diff
@@ -2309,6 +2345,72 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var sp msgjson.TradeSuspension
+	err := msg.Unmarshal(&sp)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.pendingSuspendsMtx.Lock()
+	defer dc.pendingSuspendsMtx.Unlock()
+
+	// Attempt to cancel and remove pending suspends.
+	if sched := dc.pendingSuspends[sp.MarketID]; sched != nil {
+		if !sched.Stop() {
+			return fmt.Errorf("market %s for dex %s is already suspended",
+				sp.MarketID, dc.acct.host)
+		}
+		delete(dc.pendingSuspends, sp.MarketID)
+	}
+
+	// Set the new scheduled suspend.
+	duration := time.Until(encode.UnixTimeMilli(int64(sp.SuspendTime)))
+	dc.pendingSuspends[sp.MarketID] = time.AfterFunc(duration, func() {
+		dc.pendingSuspendsMtx.Lock()
+		defer dc.pendingSuspendsMtx.Unlock()
+		dc.suspend(sp.MarketID)
+		if bookie := dc.books[sp.MarketID]; bookie != nil {
+			bookie.Reset()
+		}
+		if !sp.Persist {
+			mkt, ok := dc.marketMap[sp.MarketID]
+			if !ok {
+				log.Errorf("no market found with ID %s", sp.MarketID)
+			}
+
+			dc.tradeMtx.Lock()
+			defer dc.tradeMtx.Unlock()
+
+			// Revoke all active orders of the suspended market for the dex.
+			for _, tracker := range dc.trades {
```

Comment:
each `*trackedTrade` also has a mtx. Maybe tradeMtx trumps all of them though? Not sure.